R/cchart.R

```diff
@@ -269,19 +269,273 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
+            TitleFont = list(color = args$categories.title.font.color,
+            family = args$categories.title.font.family,
+            size = args$categories.title.font.size/1.3333),
+            AxisLine = list(Color = args$categories.line.color,
+            Width = args$categories.line.width/1.3333,
+            Style = if (isTRUE(args$categories.line.width == 0)) "None" else "Solid"),
+            MajorGridLine = list(Color = args$categories.grid.color,
+            Width = args$categories.grid.width/1.3333,
+            Style = if (isTRUE(args$categories.grid.width == 0)) "None" else "Solid"),
+            RotateLabels = isTRUE(args$categories.tick.angle == 90))
+        res$ValueAxis = list(LabelsFont = list(color = args$values.tick.font.color,
+            family = args$values.tick.font.family, size = args$values.tick.font.size/1.3333),
+            TitleFont = list(color = args$values.title.font.color,
+
+            family = args$values.title.font.family, size = args$values.title.font.size/1.3333),
+            NumberFormat = if (isTRUE(grepl("%", attr(data, "statistic")))) "0%" else "General",
+            AxisLine = list(Color = args$values.line.color,
+            Width = args$values.line.width/1.3333,
+            Style = if (isTRUE(args$values.line.width == 0)) "None" else "Solid"),
+            MajorGridLine = list(Color = args$values.grid.color,
+            Width = args$values.grid.width/1.3333,
+            Style = if (isTRUE(args$values.grid.width == 0)) "None" else "Solid"))
+
+        # We don't want to manually set axis label position
+        # if they are shown
+        if (!is.null(args$values.axis.show) && args$values.axis.show == FALSE)
+            res$ValueAxis$LabelPosition <- "None"
+        if (!is.null(args$categories.axis.show) && args$categories.axis.show == FALSE)
+            res$PrimaryAxis$LabelPosition <- "None"
+    }
+
+    # Chart-specfic parameters
+    if (chart.type %in% "Donut")
+        res$HoleSize = args$pie.inner.radius
+    if (chart.type %in% c("Bar", "Column", "Pyramid", "BarMultiColor", "ColumnMultiColor"))
+        res$GapWidth = args$bar.gap * 100
+    if (chart.type == "Line")
+        res$Smooth = isTRUE(args$shape == "Curved")
+    if (chart.type %in% c("BarMultiColor", "ColumnMultiColor", "Pyramid", "Bar Pictograph") ||
+        (chart.type == "Scatter" && !isTRUE(args$scatter.colors.as.categorical)))
+        res$ShowLegend <- FALSE
+
+
+    # There are some issues with Scatterplot exporting
+    # See RS-7154 - try master.displayr.com
+    if (chart.type %in% c("Scatter"))
+    {
+        res$BubbleSizeType = if (isTRUE(args$scatter.sizes.as.diameter)) "Width" else "Area"
+        res$BubbleScale = args$marker.size * 10
+    }
+    return(res)
+}
+
+#' @importFrom grDevices col2rgb rgb2hsv
+autoFontColor <- function (colors)
+{
+    tmp.rgb <- col2rgb(colors)
+    tmp.lum <- apply(tmp.rgb, 2, function(x) return(0.299*x[1] + 0.587*x[2] + 0.114*x[3]))
```

Comment:
Add a comment to explain where these numbers come from

Reply:
Added a comment to explain autoFontColor in a060c6a